pkg/build/builder/source_test.go

```diff
@@ -304,3 +304,72 @@ func TestCloneFromBranch(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyImageSourceFromFilesystem(t *testing.T) {
```

Comment:
This test should create the symlink in the src directory to a file in the link directory, then copy the src directory to the destination directory.  That would seem to follow more along the lines of what we would actually do?  Might just be me, but it seemed confusing the way it is done now, trying to figure out why things were being created/symlinked to other places that didn't make sense.

Reply:
@coreydaley test has been updated to be table-driven. Hopefully it's clearer.